### ISSUE TITLE
[Feature Request]: Adding Categorical Focal Loss

### ISSUE THREAD

**Issue by Frightera** [keras-team/keras#17583]
**System information**.

TensorFlow version (you are using): 2.11
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

Currently, Keras provides [built-in implementation](https://www.tensorflow.org/api_docs/python/tf/keras/losses/BinaryFocalCrossentropy) of Binary Focal Crossentropy. However, it does not provide a built-in implementation of the Categorical Focal Loss.

**Related SO questions:**
[https://stackoverflow.com/questions/56293964/categorical-focal-loss-on-keras](https://stackoverflow.com/questions/56293964/categorical-focal-loss-on-keras)
[https://stackoverflow.com/questions/65875860/how-to-use-categorial-focal-loss-for-one-hot-encoding-with-keras](https://stackoverflow.com/questions/65875860/how-to-use-categorial-focal-loss-for-one-hot-encoding-with-keras)
[https://stackoverflow.com/questions/57807175/use-of-1-a-weight-in-categorical-focal-loss](https://stackoverflow.com/questions/57807175/use-of-1-a-weight-in-categorical-focal-loss)
[https://stackoverflow.com/questions/70459884/can-sigmoidfocalcrossentropy-in-tensorflow-tf-addons-be-used-in-multiclass-cla](https://stackoverflow.com/questions/70459884/can-sigmoidfocalcrossentropy-in-tensorflow-tf-addons-be-used-in-multiclass-cla)

**Some repositories:**
[https://github.com/umbertogriffo/focal-loss-keras](https://github.com/umbertogriffo/focal-loss-keras)
[https://github.com/artemmavrin/focal-loss](https://github.com/artemmavrin/focal-loss)

**Will this change the current api? How?**
Yes, this will change the current API by adding a new built-in loss function to Keras.

**Contributing**

- Do you want to contribute a PR? (yes/no): Yes

**Comment by sampathweb** [comment]
Thanks for raising this issue.  This would be a useful feature.  Would you be able to contribute with a PR.  Please also look at reference implementation in tesorflow-addons package.

**Comment by Frightera** [comment]
@sampathweb Yes I can contribute with a PR, I'll start working on it.

**Comment by Frightera** [comment]
Closing as PR is merged now.
